aad/spring-cloud-azure-starter-active-directory/aad-resource-server-by-filter/pom.xml

```diff
@@ -35,6 +35,10 @@
       <groupId>org.springframework.boot</groupId>
       <artifactId>spring-boot-starter-security</artifactId>
     </dependency>
+    <dependency>
```

Comment:
why we need to explicit add this dependency?
anything changed from spring boot 3?
does this work without this `spring-secruity-oauth2-core` in spring boot 2.x sample?

Reply:
It's also a bug in the main branch, which will fix in the next PR.

Reply:
I remember per our discussion, the dependency should be the resource server starter, since the application acts as a resource server.

Reply:
We should think about how we could improve this. This shouldn't be the best practice.

Reply:
Opened an issue: https://github.com/Azure/azure-sdk-for-java/issues/32509